src/regionprops.jl

```diff
@@ -202,12 +246,67 @@ function cropfloe(floesimg::BitMatrix, props::DataFrame, i::Int64)
     return prefloe
 end
 
+"""
+    cropfloe(floesimg, min_row, min_col, max_row, max_col, label)
+
+Crops the floe from `floesimg` with the label `label`, returning the region bounded by `min_row`, `min_col`, `max_row`, `max_col`, and converting to a BitMatrix.
+"""
+function cropfloe(floesimg::Matrix{I}, min_row::J, min_col::J, max_row::J, max_col::J, label::I)  where {I<:Integer, J<:Integer}
+    #= 
+    Crop the floe using bounding box data in props.
+    Note: Using a view of the cropped floe was considered but if there were multiple components in the cropped floe, the source array with the floes would be modified. =#
+    prefloe = floesimg[min_row:max_row, min_col:max_col]
+    @debug "prefloe: $prefloe"
+
+    #= Remove any pixels not corresponding to that numbered floe 
+    (each segment has a different integer) =#
+    floe_area = prefloe .== label
+    @debug "mask: $floe_area"
+
+    return floe_area
+end
+
+"""
+    cropfloe(floesimg, label)
+
+Crops the floe from `floesimg` with the label `label`, adding a one pixel border of zeros and converting to a BitMatrix.
+"""
+function cropfloe(floesimg::Matrix{I}, label::I) where {I<:Integer}
+    #= Remove any pixels not corresponding to that numbered floe 
+    (each segment has a different integer) =#
+    floe_area = floesimg .== label
+    @debug "mask: $floe_area"
+
+    # Crop the floe to the size of the floe.
+    nonzero = x -> x > 0
+    rows = 2
+    cols = 1
+    row_sums = count(floe_area, dims=rows)
+    col_sums = count(floe_area, dims=cols)
+    @debug "row_sums: $row_sums, col_sums: $col_sums"
+
+    min_row = findfirst(nonzero, row_sums)[cols]
+    min_col = findfirst(nonzero, col_sums)[rows]
+    max_row = findlast(nonzero, row_sums)[cols]
+    max_col = findlast(nonzero, col_sums)[rows]
+    @debug "($min_row, $min_col), ($max_row, $max_col)"
+
+    floe_area_cropped = floe_area[min_row:max_row, min_col:max_col]
```

Comment:
Here a different method of `cropfloe` could be used making it a bit more expressive.

Reply:
How do you mean?